### PR TITLE
Add I2C support for NXP frdm_ke17z board

### PR DESCRIPTION
**Pull request by ankeXiao**
The frdm_ke17z contains only one lpi2c0 instance and tested the `samples/drivers/i2c/target_eeprom` case.

### CODE REVIEW

**Comment by ankeXiao** [comment]
Hi @hakehuang , please help to test this, thanks!

**Review by danieldegrasse — DISMISSED** [review]

**Review by MaureenHelm — DISMISSED** [review]

**Comment by hakehuang** [comment]
> Hi @hakehuang , please help to test this, thanks!

fails on board testing
```
/home/ubuntu/nxp/frdm_ke17z/temp/frdm_ke17z_twister_test_only/twister-out/frdm_ke17z/samples/sensor/accel_polling/sample.sensor.accel_polling/handler.log
*** Booting Zephyr OS build v3.6.0-6332-g19970cb533da ***
sensor: device fxos8700@1c not ready.
```

File: boards/nxp/frdm_ke17z/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# FRDM-KE17Z board
+
+# Copyright 2024 NXP
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_FRDM_KE17Z
+
+config FXOS8700_DRDY_INT1
+	default y
```
Comment (hakehuang) [inline boards/nxp/frdm_ke17z/Kconfig.defconfig:9]: this is DNP in revB board, so we should not enable it

Comment (ankeXiao) [reply boards/nxp/frdm_ke17z/Kconfig.defconfig:9]: Thanks for checking and testing, I will drop the sensor related configuration

File: samples/drivers/i2c/target_eeprom/sample.yaml [hunk 2]
```diff
@@ -6,4 +6,6 @@ tests:
     filter: dt_nodelabel_enabled("target_eeprom")
     platform_allow:
       - lpcxpresso55s69/lpc55s69/cpu0
+      - frdm_ke17z
+      - frdm_ke17z512
     harness: TBD
```
Comment (hakehuang) [inline samples/drivers/i2c/target_eeprom/sample.yaml:11]: if you enable it  please change harness like below
```
   harness: console
    harness_config:
      type: multi_line
      regex:
        - ".*: Cipher Sample"
        - ".*: ECB mode ENCRYPT - Match"
        - ".*: ECB mode DECRYPT - Match"
        - ".*: CBC mode ENCRYPT - Match"
        - ".*: CBC mode DECRYPT - Match"
        - ".*: CTR mode ENCRYPT - Match"
        - ".*: CTR mode DECRYPT - Match"
```

Comment (ankeXiao) [reply samples/drivers/i2c/target_eeprom/sample.yaml]: Thanks for reminding, I have updated the sample.yaml to support them

**Comment by ankeXiao** [comment]
Hi All, please help to review this PR again, tested `samples/drivers/i2c/target_eeprom` to replace the fxos8700 sensor sample as it is DNP on RevB.

**Review by hakehuang — APPROVED** [review]

**Comment by hakehuang** [comment]
board testing PASS.

```
2024-07-05 16:19:13,401 - twister - DEBUG - DEVICE: *** Booting Zephyr OS build v3.6.0-6332-g3b55f4face57 ***
2024-07-05 16:19:13,402 - twister - DEBUG - DEVICE: i2c target sample
2024-07-05 16:19:13,402 - twister - DEBUG - HARNESS:Console:EXPECTED(1/3):'i2c target sample'
2024-07-05 16:19:13,405 - twister - DEBUG - DEVICE: i2c target driver registered
2024-07-05 16:19:13,405 - twister - DEBUG - HARNESS:Console:EXPECTED(2/3):'i2c target driver registered'
2024-07-05 16:19:14,407 - twister - DEBUG - DEVICE: i2c target driver unregistered
2024-07-05 16:19:14,407 - twister - DEBUG - HARNESS:Console:EXPECTED(3/3):'i2c target driver unregistered'
2024-07-05 16:19:14,408 - twister - DEBUG - run status: frdm_ke17z/samples/drivers/i2c/target_eeprom/sample.drivers.i2c.target.kinetis passed
```

**Review by MaureenHelm — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**